File: fbi_engine_analyse/src/com/fr/bi/cal/analyze/report/report/widget/chart/newstyle/filter/condition/SingleFilter.java
package com.fr.bi.cal.analyze.report.report.widget.chart.newstyle.filter.condition;

import com.fr.bi.cal.analyze.report.report.widget.chart.newstyle.filter.FilterValueFactory;
import com.fr.bi.cal.analyze.report.report.widget.chart.newstyle.filter.IFilter;
import com.fr.json.JSONArray;
import com.fr.json.JSONException;
import com.fr.json.JSONObject;

/**
 * Created by AstronautOO7 on 2016/9/19.
 */
public class SingleFilter implements IFilter {

    private IFilter filterValue;

    public SingleFilter(JSONObject jo) throws JSONException{
        filterValue = FilterValueFactory.parseFilter(jo);
    }

    @Override
    public JSONArray getFilterResult(JSONArray array) throws JSONException {
        return this.filterValue.getFilterResult(array);
    }
}
 

File: web/fbi_web/src/com/fr/bi/web/js/data/filter/condition/single.filter.js
;
!(function(){
    BI.SingleFilter = function(filter){
        this.filterValue = BI.FilterValueFactory.parseFilterValue(filter);
    };
    BI.SingleFilter.prototype = {
        constructor: BI.SingleFilter,

        getFilterResult: function(array) {
            return this.filterValue.getFilterResult(array);
        }
    }
})();